Navigation from experimental feature to the related page

davidegiacometti opened the pull request:
## Summary of the pull request

Added navigation from experimental feature to the related page.

## References and relevant issues

#2909

## Detailed description of the pull request / Additional comments

- Changed `NavConfig.jsonc`: added page key/parameter for opening the experimental feature related page.
- Added a button to open the experimental feature page in experimental feature settings.
- Handled navigation in pages in order to show QuickstartPlayground and FileExplorer view.

![dev](https://github.com/user-attachments/assets/7f37b1ac-1073-4e94-9fb6-568bcf46e2a3)

## Validation steps performed

- Manually tested the navigation for the 4 existing experimental feature.
- Validated that the "Open" button is rendered as expected after switching dark/light theme.

## PR checklist
- [x] Closes #2909
- [ ] Tests added/passed
- [ ] Documentation updated

AmelBawa-msft commented on settings/DevHome.Settings/Views/ExperimentalFeaturesPage.xaml:
```diff
@@ -28,13 +28,35 @@
+                                    Orientation="Horizontal"
+                                    Spacing="16"
+                                    Grid.Column="1">
+                                    <ToggleSwitch IsOn="{x:Bind IsEnabled, Mode=OneWay}">
```
Wondering if we should use the `ActionIcon` property in `SettingsCard` instead. That ensures we inherit the styles from the SettingsCard control (forward compatible). But that will make the entire button clickable.
```xml
<ctControls:SettingsCard
    ...
    ActionIcon="{ui:FontIcon Glyph=&#xE8A7;}"
    IsClickEnabled="True">
```

Related links:
- https://github.com/microsoft/WinUI-Gallery/blob/cb323e1a5dff53e482ce577b48ca311cfe8d00cd/WinUIGallery/SettingsPage.xaml#L120-L128

AmelBawa-msft commented on src/NavConfig.jsonc:
```diff
@@ -66,7 +66,10 @@
-      ]
+      ],
+      "openPage": {
+        "key": "DevHome.Customization.ViewModels.MainPageViewModel"
```
Just thinking out loud for the future. If we implement a protocol processor in Dev Home to help us navigate between pages and activate Dev Home with a protocol to perform certain actions (e.g. something like ms-devhome://{page}?{params}). We may be able to use that here as well and maintain a single implementation. Or both can also co-exist as long as the engine processing the two interfaces is the same.

#NoActionRequired

krschau reviewed (changes requested):
We should have caught this in the spec, so I apologize for asking for this change, but instead of the "open in new window" button we should be using the IsClickEnabled/Command built into the settings card to do the navigation.
https://github.com/microsoft/devhome/issues/2909#issuecomment-2289494118

davidegiacometti commented:
Don't worry Kristen, changes make sense and also look better. 😃 

![image](https://github.com/user-attachments/assets/66b775d4-2396-40de-b465-8d8235357f20)

I don't really like the misaligned toggle switches for disabled features 😞
WDYT?

krschau commented:
> I don't really like the misaligned toggle switches for disabled features 😞 WDYT?

Ooh agreed, I don't love that. @TWhiteDesigns, thoughts? 

I'm not sure where on the spectrum of "kind of annoying" to "shockingly difficult" it would be to nudge the toggle over to match in cases where the > icon doesn't show. If it's not bad, I'd suggest that.

davidegiacometti commented:
Easy to do using `DataTriggerBehavior + ChangePropertyAction` 😃 
Here is a screen of the proposed solution.

![image](https://github.com/user-attachments/assets/8c849711-dc7e-493f-8720-bb46227f904f)

krschau commented:
> Easy to do using `DataTriggerBehavior + ChangePropertyAction` 😃 Here is a screen of the proposed solution.

@davidegiacometti I think this looks great, let's do it!

davidegiacometti commented:
I have addressed the feedback and updated the screenshot in the PR description.

A small nit I noticed on Windows 10 related to experimental feature: Quiet background processes and File Explorer source control integrations aren't available. The first experimental feature is totally hidden, but, the second one is visible only in the list.

krschau commented:
> I have addressed the feedback and updated the screenshot in the PR description.
> 
> A small nit I noticed on Windows 10 related to experimental feature: Quiet background processes and File Explorer source control integrations aren't available. The first experimental feature is totally hidden, but, the second one is visible only in the list.

Good catch. Quiet background processes showing up was fixed by #3545, but I opened #3679 to fix File Explorer integration.
